### PR TITLE
Support both HTTP and HTTPS & Update to ES v7

### PR DESCRIPTION
**Pull request by spencerdcarlson**
## Run HTTPS
1. Set your .env to run HTTPS (see env.example for reference)
1. Run `./bin/dev/start` to spin up ES and Kibana
1. Run `./bin/dev/clean` to teardown all docker assets

## Run HTTP
1. Change .env to run HTTP (see env.example for reference)
1. Run `./bin/dev/start` to spin up ES and Kibana
1. Run `./bin/dev/clean` to teardown all docker assets

### CODE REVIEW

File: elastic.go [hunk 1]
```diff
@@ -46,17 +46,74 @@ func SetupElastic() {
 	bulkProc["bulk"] = proc
 }
 
+func elasticClientFactory(isInsecure bool) *elastic.Client {
```
Comment (spencerdcarlson) [inline elastic.go:49]: Going to change this to come from an ENV variable

File: docker/es/certs/instances.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      - localhost
+    ip:
+      - 127.0.0.1
+
+  - name: kibana
+    dns:
+      - kibana
+      - localhost
+
+#  - name: es02
```
Comment (brianbroderick) [inline docker/es/certs/instances.yml:14]: do we need these comments or can we delete?

File: bin/dev/es/es_functions [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+function es_is_ready() {
+    local ES_HOST=${1:-https://es01:9200}
+    local EXEC_CONTAINER=${2:-kibana} # Any docker container that has access to the ca.crt
+    docker exec $EXEC_CONTAINER \
+      curl -sIfo /dev/null \
+      --cacert $CERTS_DIR/ca/ca.crt \
+      -u elastic:$ELASTIC_PASSWORD \
+      $ES_HOST
+}
+
```
Comment (brianbroderick) [inline bin/dev/es/es_functions:62]: not sure what these comments are for?

File: bin/dev/kibana/kibana_functions [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+function kibana_is_ready() {
+  local KIBANA_HOST=${1:-https://kibana:5601}
+  local EXEC_CONTAINER=${2:-es01} # Any docker container that has access to the ca.crt
+  docker exec $EXEC_CONTAINER \
+    curl -Isfo /dev/null \
+    -m 2 \
+    --cacert $CERTS_DIR/ca/ca.crt \
+    $KIBANA_HOST/login?next=%2F
+}
+
```
Comment (brianbroderick) [inline bin/dev/kibana/kibana_functions:27]: same

File: elastic_test.go [hunk 5]
```diff
@@ -196,14 +196,14 @@ func getRecord(t *testing.T, wait time.Duration, username string) float64 {
 		panic(err)
 	}
 
-	if result.Hits.TotalHits > 0 {
-		fmt.Printf("Found a total of %d record(s)\n", result.Hits.TotalHits)
+	if result.Hits.TotalHits.Value > 0 {
```
Comment (brianbroderick) [inline elastic_test.go:199]: lets change fmt to use logit instead.

File: elastic_test.go [hunk 6]
```diff
@@ -255,14 +255,14 @@ func getRecordWithTempTable() int64 {
...
+		fmt.Printf("Found a total of %d record(s)\n", result.Hits.TotalHits.Value)
```
Comment (brianbroderick) [inline elastic_test.go:259]: lets change fmt to use logit instead

File: stats_test.go [hunk 7]
```diff
@@ -69,14 +69,14 @@ func getStatsRecord(t *testing.T, wait time.Duration, app string) int64 {
 		panic(err)
 	}
 
-	if result.Hits.TotalHits > 0 {
-		fmt.Printf("Found a total of %d stat record(s)\n", result.Hits.TotalHits)
+	if result.Hits.TotalHits.Value > 0 {
+		fmt.Printf("Found a total of %d stat record(s)\n", result.Hits.TotalHits.Value)
```
Comment (brianbroderick) [inline stats_test.go:73]: lets change fmt to use logit instead

**Pull request merged**